soccer/gameplay/situational_play_selection.py

```diff
@@ -300,47 +319,67 @@ def isPileup(self):
 
     ##Returns if we are in the specified situation without regard to capitalization
     #If the check variable is true, it will check if the situation exists and throw an
-    #exception is it does now.
-    def isSituation(self, situation, check=False):
-        up = situation.upper()
-
-        if (check):
-            found = False
-            for g in self.Situation:
-                if (up == g):
-                    found = True
-                    break
-            if not found:
-                raise Exception("Passed situation " + situation + " / " + up +
-                                " is not an existing situation")
-
-        if (self.currentSituaion.name == up):
-            return True
+    #exception is it does now
+    # @param situation a situation as a string or a enum 
+    # @param check Strings will be checked if they are real if True
+    def isSituation(self, situation, check=True):
+
+        if (isinstance(situation, str)):
+            up = situation.upper()
+
+            if (check):
+                found = False
+                for g in self.Situation:
+                    if (up == g):
+                        found = True
+                        break
+                if not found:
+                    raise ValueError("Passed situation " + situation + " / " +
+                                     up + " is not an existing situation")
+
+            if (self.currentSituation.name == up):
+                return True
+            else:
+                return False
+
+        elif (isinstance(situation, self.Situation)):
+            return situation == self.currentSituation
         else:
-            return False
+            raise TypeError("isSituation only takes strings and enums")
 
-    ##Update determining if we want to preempt the current play or not
-    #
-    # Preemption is still an open question but this is a prototype of
-    # of how a non-invasive preemption system might work
-    #
+    ##Returns true if we are in any of the situations in the passed list
+    #Will take enums or 
+    def isSituations(self, situations, check=False):
+        for g in situations:
+            if (self.isSituation(g, check=check)):
+                return True
+
+        return False
+
+    ##
+    # 
+    # Update determining if we want to preempt the current play or not
+    #   
     def updatePreempt(self):
```

Comment:
I would prefer this have some interaction with the play itself. i.e. `if currentPlay.tryPreempt(self.currentSituation): # Try preempt`. But I'm open to changing my mind on this.

Reply:
Currently planning on adding
```
    def try_preempt(self):
        self.terminate()
        return True
```
to standard play that can be overridden if you want more complex functionality, which will also remove preempt logic from execute_running and change update_preempt to be actually calling for preempts rather than just updating a value.

Reply:
That sounds good to me.